src/common/src/array/map_array.rs

```diff
@@ -0,0 +1,294 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::cmp::Ordering;
+use std::fmt::{self, Debug, Display};
+
+use itertools::Itertools;
+use risingwave_common_estimate_size::EstimateSize;
+use risingwave_pb::data::{PbArray, PbArrayType};
+
+use super::{
+    Array, ArrayBuilder, ArrayImpl, ArrayResult, DatumRef, ListArray, ListArrayBuilder, ListRef,
+    ListValue, MapType, ScalarRefImpl, StructArray,
+};
+use crate::bitmap::Bitmap;
+use crate::types::{DataType, Scalar, ToText};
+
+#[derive(Debug, Clone, EstimateSize)]
+pub struct MapArrayBuilder {
+    inner: ListArrayBuilder,
+}
+
+impl ArrayBuilder for MapArrayBuilder {
+    type ArrayType = MapArray;
+
+    #[cfg(not(test))]
+    fn new(_capacity: usize) -> Self {
+        panic!("please use `MapArrayBuilder::with_type` instead");
+    }
+
+    #[cfg(test)]
+    fn new(capacity: usize) -> Self {
```

Comment:
Where is this method used?

Reply:
In some tests. I tried to deprecate this, but it's not easy. Thus delayed.